tf_big/cc/ops/big_ops.cc

```diff
@@ -1,32 +1,70 @@
 #include "tensorflow/core/framework/op.h"
+#include "tensorflow/core/framework/common_shape_fns.h"
 #include "tensorflow/core/framework/shape_inference.h"
 
 REGISTER_OP("BigImport")
     .Attr("dtype: {int32, string}")
     .Input("in: dtype")
     .Output("val: variant")
-    .SetIsStateful();
+    .SetIsStateful()
+    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
+        ::tensorflow::shape_inference::ShapeHandle output;
+        TF_RETURN_IF_ERROR(c->WithRankAtMost(c->input(0), 2, &output));
+        c->set_output(0, output);
+        return ::tensorflow::Status::OK();
+    });
 
 REGISTER_OP("BigExport")
     .Attr("dtype: {int32, string}")
     .Input("val: variant")
     .Output("out: dtype")
-    .SetIsStateful();
+    .SetIsStateful()
+    .SetShapeFn(::tensorflow::shape_inference::UnchangedShape);
 
 REGISTER_OP("BigAdd")
+    .Input("val0: variant")
     .Input("val1: variant")
-    .Input("val2: variant")
     .Output("res: variant")
-    .SetIsStateful();
+    .SetIsStateful()
+    .SetShapeFn([](::tensorflow::shape_inference::InferenceContext* c) {
+        ::tensorflow::shape_inference::ShapeHandle val0 = c->input(0);
+        ::tensorflow::shape_inference::ShapeHandle val1 = c->input(1);
+        ::tensorflow::shape_inference::ShapeHandle res;
+        TF_RETURN_IF_ERROR(c->Merge(val0, val1, &res));
```

Comment:
Having trouble understanding what this `Merge` function does. Simple explanation?

Reply:
It essentially makes sure that the shapes of `val0` and `val1` are the same, filling in missing values from one with concrete values from the other. These filled-in shapes are then stored in `res`. More [details in the TF doc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/shape_inference.h#L230-L251).

Reply:
Thanks!